pkg/codegen/nodejs/gen_program_expressions.go

```diff
@@ -177,7 +177,7 @@ func (g *generator) GenForExpression(w io.Writer, expr *model.ForExpression) {
 
 	if expr.Key != nil {
 		// TODO(pdg): grouping
-		g.Fgenf(w, ".reduce((__obj, %s) => { ...__obj, [%.v]: %.v })", reduceParams, expr.Key, expr.Value)
+		g.Fgenf(w, ".reduce((__obj, %s) => ({ ...__obj, [%.v]: %.v }))", reduceParams, expr.Key, expr.Value)
```

Comment:
Small fix for this expression where it was missing the extra parenthesis around the returned object, otherwise it is invalid TypeScript